tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/Controllers/AccountControllerTests.cs

```diff
@@ -0,0 +1,112 @@
+            var actual = await sut.Logout(expectedLogoutId);
+            
+            actual.Should().BeEquivalentTo(new RedirectResult(expectedPostDirectUrl));
+        }
+
+        [Test]
+        public async Task Logout_WhenEmptyLogoutViewModel_ReturnsRedirectResult()
+        {
+            const string expectedPostDirectUrl = "localhost";
+
+            var logoutServiceMock = new Mock<ILogoutService>();
+            logoutServiceMock.Setup(x => x.SignOutAsync(It.IsAny<LogoutViewModel>()))
+                .Returns(Task.CompletedTask);
+            logoutServiceMock.Setup(x => x.GetPostLogoutRedirectUri(It.IsAny<string>()))
+                .ReturnsAsync(expectedPostDirectUrl);
+
+            using var sut = new AccountControllerBuilder()
+                .WithLogoutService(logoutServiceMock.Object)
+                .Build();
+
+            LogoutViewModel expectedLogoutViewModel = new LogoutViewModel();
+            var actual = await sut.Logout(expectedLogoutViewModel);
+            
+            actual.Should().BeEquivalentTo(new RedirectResult(expectedPostDirectUrl));
+        }
+
+        [Test]
+        public async Task Logout_WhenEmptyLogoutViewModel_LogoutServiceSignOutCalledOnce()
+        {
+            const string expectedPostDirectUrl = "localhost";
+
+            var logoutServiceMock = new Mock<ILogoutService>();
+            logoutServiceMock.Setup(x => x.SignOutAsync(It.IsAny<LogoutViewModel>()))
+                .Returns(Task.CompletedTask);
+            logoutServiceMock.Setup(x => x.GetPostLogoutRedirectUri(It.IsAny<string>()))
+                .ReturnsAsync(expectedPostDirectUrl);
+
+            using var sut = new AccountControllerBuilder()
+                .WithLogoutService(logoutServiceMock.Object)
+                .Build();
+
+            LogoutViewModel expectedLogoutViewModel = new LogoutViewModel();
+            await sut.Logout(expectedLogoutViewModel);
+
+            logoutServiceMock.Verify(x => x.SignOutAsync(It.Is<LogoutViewModel>(actual => actual.Equals(expectedLogoutViewModel))), Times.Once);
+        }
+
+        [Test]
+        public async Task Logout_WhenEmptyLogoutViewModel_LogoutServiceGetPostLogoutRedirectUriCalledOnce()
+        {
+            const string expectedPostDirectUrl = "localhost";
+
+            var logoutServiceMock = new Mock<ILogoutService>();
+            logoutServiceMock.Setup(x => x.SignOutAsync(It.IsAny<LogoutViewModel>()))
+                .Returns(Task.CompletedTask);
+            logoutServiceMock.Setup(x => x.GetPostLogoutRedirectUri(It.IsAny<string>()))
+                .ReturnsAsync(expectedPostDirectUrl);
+
+            using var sut = new AccountControllerBuilder()
+                .WithLogoutService(logoutServiceMock.Object)
+                .Build();
+
+            await sut.Logout(new LogoutViewModel());
+
+            logoutServiceMock.Verify(x => x.GetPostLogoutRedirectUri(
+                It.Is<string>(actual => actual == null)), Times.Once);
+        }
```

Comment:
Could you explain your thoughts on this test? Unless I've missed something won't `GetPostLogoutRedirectUri` only ever be called once regardless of the state `logoutViewModel` (except the `null` case)?

Reply:
The idea here is to test dependencies used within the method. So when I call that method I want to ensure that my dependency is getting called only once & how I expect it. I didn't want to do it in every test because if the method changes every test would fail. At least this way there is the potential for only this one that would fail.

This was a new concept for me in unit tests. The previous tech lead was keen on this. In the past I have only ever tested input output. But one could class the dependency as input I guess.